aws/resource_aws_s3_bucket.go

```diff
@@ -1589,11 +1589,15 @@ func WebsiteEndpoint(bucket string, region string) *S3Website {
 func WebsiteDomainUrl(region string) string {
 	region = normalizeRegion(region)
 
-	// New regions uses different syntax for website endpoints
-	// http://docs.aws.amazon.com/AmazonS3/latest/dev/WebsiteEndpoints.html
+	// Different regions have different syntax for website endpoints
+	// https://docs.aws.amazon.com/AmazonS3/latest/dev/WebsiteEndpoints.html
+	// https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_website_region_endpoints
 	if isOldRegion(region) {
 		return fmt.Sprintf("s3-website-%s.amazonaws.com", region)
 	}
+	if isChineseRegion(region) {
```

Comment:
Hi @antonio-osorio 👋 Thanks for submitting this. Instead of performing our own custom logic here, can you please switch this to using the AWS Go SDK `endpoints` package?

```suggestion
	if partition, ok := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region); ok && partition.ID() == endpoints.AwsCnPartitionID {
```

This will automatically update the AWS China regions with AWS Go SDK updates as well as attempt regular expression lookups if the region name cannot be directly found. Thanks!

I have submitted an upstream feature request in the AWS to directly provide the DNS suffix information, which can help prevent us from hardcoding these cases in the future: https://github.com/aws/aws-sdk-go/issues/2710

Reply:
Updated.